src/main/java/subway/auth/ui/AuthenticationPrincipalArgumentResolver.java

```diff
@@ -23,7 +27,9 @@ public boolean supportsParameter(MethodParameter parameter) {
     // parameter에 @AuthenticationPrincipal이 붙어있는 경우 동작
     @Override
     public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer, NativeWebRequest webRequest, WebDataBinderFactory binderFactory) {
-        // TODO: 유효한 로그인인 경우 LoginMember 만들어서 응답하기
-        return null;
+        String accessToken = AuthorizationExtractor.extract((HttpServletRequest) webRequest.getNativeRequest());
+        if (accessToken == null)
+            return null;
+        return LoginMember.of(authService.checkInvalidToken(accessToken));
     }
```

Comment:
> 토큰 검증이나, 맴버 정보가 필요한 부분에 전부 @AuthenticationPrincipal를 이용하여서 맴버 객체를 가져오는 방법을 사용했습니다. 맴버가 필요없는 부분에는 @AuthenticationPrincipal대신 인터셉터를 이용해서 검증하는게 옳을까요?

AuthenticationPrincipal라기보다는 리졸버와 인터셉터의 차이로 보는게 좋을 것 같네요.

AuthenticationPrincipalArgumentResolver가 구현한 HandlerMethodArgumentResolver를 보도록 하죠.
```java
/**
 * Strategy interface for resolving method parameters into argument values in
 * the context of a given request.
 *
 * @author Arjen Poutsma
 * @since 3.1
 * @see HandlerMethodReturnValueHandler
 */
public interface HandlerMethodArgumentResolver {
```
ArgumentResolver라는 이름에서도 보이다시피 맴버가 필요하지 않으면 사용하지 않는게 바람직하다고 생각합니다. 다만, 이번 과제 기준으로만 봤을 때는 인증과 인가 어느게 필요한지 중요할 것 같아요.

```java
    @PutMapping("/members/{id}")
    public ResponseEntity<MemberResponse> updateMember(@PathVariable Long id, @RequestBody MemberRequest param);
    @DeleteMapping("/members/{id}")
    public ResponseEntity<MemberResponse> deleteMember(@PathVariable Long id);
```
아무런 확인 없이 사용자를 지우고 수정할 수 있네요. 강의에서 다뤘을 수 있을걸로 보이는데 /members/me와의 차이를 두기 위해 뒀거나 관리를 위한 API로 인증이 아닌 인가로 수정/삭제 권한을 가진 사용자인 확인하는 걸 목적으로 둔 api로 보입니다.

--- 

AuthenticationPrincipal가 필요없다고 본 곳이 `/favorites/{id}`가 맞을까요? 맞다면 두가지 질문 드릴게요.
이 API의 사용처는 사용자인가요? 아니면 관리자 페이지를 사용하는 관리자일까요?
그에 따라 인증, 인가 중 어떤 처리가 필요할까요? 
(인증이 필요하다면 지금은 다른 사용자가 즐겨찾기 id만 알면 남의 즐겨찾기를 지울 수 있지 않을까요?  🤔)

Reply:
과제의 요구사항을 봤을 때 `@AuthenticationPrincipal과 AuthenticationPrincipalArgumentResolver을 활용`이나 별도로 인가에 대한 요구가 없는 걸로 보여 지금으로도 충분해 보입니다. 
하나의 서비스를 만들어보고 정말 꼼꼼하게 적용해보는게 가장 좋지만 강의 진행 속도를 봤을 때 요구사항에 맞춰 한정된 부분에 대해서 잘 적용하는걸로 마무리 짓고, 이번 과제의 강의에 있는 spring의 구조([테스트로 배우는 스프링 MVC Config](https://edu.nextstep.camp/s/bb4PXLji/ls/MIOwm2me)에 있는)와 인증, 인가, 세션, JWT, Oauth, ...에 대해서 다시 한번 정리해보는 걸 추천 드립니다. 😉

Reply:
LoginMember가 필요없다고 생각한 부분은 @DeleteMapping("/favorites/{id}")
    public ResponseEntity<Void> deleteFavorite(@AuthenticationPrincipal LoginMember loginMember, @PathVariable Long id)  이 부분이었습니다. 이 부분에서 member를 이용하지 않는데 저렇게 토큰을 인증하는 용도로 @AuthenticationPrincipal LoginMember loginMember를 사용하는게 옳은건가? 하는 의문이 들어서 질문을 했었습니다.

이 댓글을 달면서 코드를 보니 deleteFavorite의 경우 토큰의 유효성만 판별하고 지우는 유저가 favarite의 주인이 맞는지에 대해서는 검증하지 않네요, 이부분도 수정해보겠습니다.